### PR TITLE
Cde design system

### PR DESCRIPTION
**Pull request by edlu77**
Added design system and updated styles for landing page, create vis page, and visualization page

### CODE REVIEW

**Review by LibbyUX — CHANGES_REQUESTED** [review]
Hi @edlu77, thank you so much for getting this started! It's looking good!

Some requests for this preview:
- I'm looking for updates in our design system to build this nav header:
  - I created a list of design system updates for all nav headers on this issue to help organize what is needed in the design system first before implementing in the CDE: #767
- Is it possible to get this to stick at the top of the page on scroll? This is functionality needed for all nav headers in the design system.
- I would like to see the updated HuBMAP apps sidenav implemented as well on click of the Apps icon button.

Thank you so much! Please don't hesitate to reach out if you have questions!

image

**Comment by edlu77** [comment]
@LibbyUX I've made the header updates in a separate PR: https://github.com/hubmapconsortium/hra-ui/pull/778

Please comment there with any feedback. Thanks!

File: libs/design-system/app-logos/src/lib/app-logos.component.scss [hunk 1]
```diff
@@ -47,6 +47,17 @@
       letter-spacing: var(--sys-label-medium-tracking);
       color: var(--sys-on-secondary-fixed);
     }
+
+    // Temporary
+    .beta-description {
```
Comment (axdanbol) [inline libs/design-system/app-logos/src/lib/app-logos.component.scss:52]: I wonder if there is a better way to do this or if we should ask Libby not to have 'Beta' as a description? There is a major risk that we forget to remove this when changing to a proper description.

File: apps/cde-ui/src/app/pages/landing-page/landing-page.component.theme.scss [hunk 2]
```diff
@@ -1,54 +1,7 @@
-@use '@angular/material' as mat;
-@use '../../../styles/cta' as cta;
-@mixin color($theme) {
-  cde-landing-page {
-    .hra-btn.cta-flat {
-      color: mat.get-theme-color($theme, primary, 700);
+cde-landing-page {
```
Comment (axdanbol) [inline apps/cde-ui/src/app/pages/landing-page/landing-page.component.theme.scss:1]: Please move the content into `landing-page.component.scss` and delete this file

File: apps/cde-ui/src/app/pages/create-visualization-page/create-visualization-page.component.theme.scss [hunk 3]
```diff
@@ -1,15 +1,6 @@
 cde-create-visualization-page {
```
Comment (axdanbol) [inline apps/cde-ui/src/app/pages/create-visualization-page/create-visualization-page.component.theme.scss:1]: Move content into `create-visualization-page.component.scss` and delete this file

File: apps/cde-ui/src/styles.scss [hunk 4]
```diff
@@ -1,5 +1,7 @@
 @use 'light-theme';
 @use 'theming';
+@use '../src/app/pages/create-visualization-page/create-visualization-page.component.theme.scss' as create-visualization;
```
Comment (axdanbol) [inline apps/cde-ui/src/styles.scss:3]: Remove theme files

File: libs/cde-visualization/src/lib/components/histogram/histogram.component.html [hunk 5]
```diff
@@ -20,10 +20,16 @@
           [cdkConnectedOverlayOpen]="infoOpen"
           [cdkConnectedOverlayOffsetX]="5"
         >
-          <div>
-            The graph shows a histogram of cell-to-nearest-anchor cell distance distributions categorized by each cell
-            type in the dataset.
-          </div>
+          <hra-tooltip-card
+            small
+            [content]="[
```
Comment (axdanbol) [inline libs/cde-visualization/src/lib/components/histogram/histogram.component.html:25]: Move content into a readonly variable on the component class and use that instead of embedding complex objects in the html

File: libs/cde-visualization/src/lib/components/metadata/metadata.component.html [hunk 6]
```diff
@@ -20,7 +20,16 @@
           [cdkConnectedOverlayOpen]="tooltipOpen"
           [cdkConnectedOverlayOffsetX]="5"
         >
-          <div>Visualization metadata for the sample dataset. Sample files may be viewed in Google Sheets.</div>
+          <hra-tooltip-card
+            small
+            [content]="[
```
Comment (axdanbol) [inline libs/cde-visualization/src/lib/components/metadata/metadata.component.html:25]: Move content into a readonly variable on the component class and use that instead of embedding complex objects in the html

File: libs/cde-visualization/src/lib/components/violin/violin.component.html [hunk 7]
```diff
@@ -16,7 +16,16 @@
     [cdkConnectedOverlayOpen]="infoOpen"
     [cdkConnectedOverlayOffsetX]="5"
   >
-    <div>Violin plot</div>
+    <hra-tooltip-card
+      small
+      [content]="[
```
Comment (axdanbol) [inline libs/cde-visualization/src/lib/components/violin/violin.component.html:21]: Move content into a readonly variable on the component class and use that instead of embedding complex objects in the html

**Review by axdanbol — CHANGES_REQUESTED** [review]

**Comment by LibbyUX** [comment]
@axdanbol, @edlu77,

Regarding the Software Status Indicator, this approved design articulates where we are at with our software. When an update is required, I can submit individual issues per application using it. We need 3 sizes for our various UIs. Let me know if you have any questions!

# Examples in approved UIs

### Resources
- #763 
- [Design System Repo](https://www.figma.com/design/gQEMLugLjweDvbsNNUVffD/HRA-Design-System-Repository?node-id=6283-32615&t=qeGxvX170NO0Ew4b-4)
- [HRA Components](https://www.figma.com/design/BCEJn9KCIbBJ5MzqnojKQp/HRA-Components?node-id=1409-13182)

## FTU Explorer nav header
- Using small variant:

Screenshot 2024-10-17 at 12 12 49 PM

## App Library
- Using large variant:

Screenshot 2024-10-17 at 12 14 02 PM

File: libs/design-system/app-logos/src/lib/app-logos.component.scss (hunk 1 above)
Comment (edlu77) [reply libs/design-system/app-logos/src/lib/app-logos.component.scss:52]: I updated the header in #778 to use the software status indicator, so this won't be needed anymore after that is merged in.

File: apps/cde-ui/src/app/components/file-upload/file-upload.component.scss [hunk 8]
```diff
@@ -1,55 +1,29 @@
...
-  .upload-success {
-    height: 2.5rem;
-    display: flex;
-    align-items: center;
-    padding: 0 1rem;
-    background: var(--sys-surface-container);
-    gap: 0.5rem;
-    border-radius: 1rem;
-    font: var(--sys-label-large);
+  // .upload-success {
```
Comment (axdanbol) [inline apps/cde-ui/src/app/components/file-upload/file-upload.component.scss:2]: Remove commented out code please

File: apps/cde-ui/src/app/components/file-upload/file-upload.component.spec.ts [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
 describe('FileUploadComponent', () => {
   const loader = jest.fn();
-  const mockFiles = {
-    0: {
-      name: 'nodes.csv',
-      type: 'text/csv',
-    } as File,
-    length: 1,
-  } as unknown as FileList;
+  // const mockFiles = {
```
Comment (axdanbol) [inline apps/cde-ui/src/app/components/file-upload/file-upload.component.spec.ts:9]: Remove commented out code please

**Review by axdanbol — CHANGES_REQUESTED** [review]
Cleanup some commented out code

File: apps/cde-ui/src/app/components/file-upload/file-upload.component.spec.ts [hunk 10]
```diff
@@ -21,29 +14,12 @@ describe('FileUploadComponent', () => {
     });
   });
 
-  it('should load', async () => {
```
Comment (axdanbol) [inline apps/cde-ui/src/app/components/file-upload/file-upload.component.spec.ts:24]: Why did you remove the tests for file loading?

**Review by axdanbol — CHANGES_REQUESTED** [review]
I'm still seeing quite a bit of commented out code. Also please resolve merge conflicts

**Comment by edlu77** [comment]
@LibbyUX I've merged all the changes from #788 into this branch. Please review this version, thanks!

**Review by LibbyUX — APPROVED** [review]
Amazing, @edlu77 - Thank you so much!

File: libs/cde-visualization/src/lib/cde-visualization/cde-visualization.component.spec.ts [hunk 11]
```diff
@@ -109,41 +109,43 @@ describe('CdeVisualizationComponent', () => {
...
-    });
-
-    const fileSaver = TestBed.inject(FileSaverService);
-    const fileSaveSpy = jest.spyOn(fileSaver, 'saveCsv').mockReturnValue(undefined);
-
-    const downloadEdgesButton = screen.getByText('Edges');
-    await userEvent.click(downloadEdgesButton);
-    expect(fileSaveSpy).toHaveBeenCalledWith(instance.addEdgeHeaders(instance.loadedEdges()), 'edges.csv');
-  });
+  // Test disabled until download nodes button is readded
```
Comment (axdanbol) [inline libs/cde-visualization/src/lib/cde-visualization/cde-visualization.component.spec.ts:112]: Will be fixed by #772

**Review by axdanbol — APPROVED** [review]
